src/rez/wheel.py

```diff
@@ -0,0 +1,527 @@
+    """Convenience function to below functions
+
+    Arguments:
+        names (list): pip-formatted package names, e.g. six=1.12
+        prefix (str, optional): Absolute path to destination repository
+        no_deps (bool, optional): Do not install dependencies,
+            equivalent to pip --no-deps
+        release (bool, optional): Install onto REZ_RELEASE_PACKAGES_PATH
+        variants (list, optional): Override variants detected by WHEEL
+
+    """
+
+    assert prefix is None or isinstance(prefix, _basestring), (
+        "%s was not str" % prefix)
+    assert isinstance(names, (tuple, list)), "%s was not list or tuple" % names
+
+    tempdir = tempfile.mkdtemp(suffix="-rez", prefix="pip-")
+
+    distributions = download(
+        names,
+        tempdir=tempdir,
+        no_deps=no_deps,
+    )
+
+    packagesdir = prefix or (
+        config.release_packages_path if release
+        else config.local_packages_path
+    )
+
+    new, existing = list(), list()
+    for dist in distributions:
+        package = convert(dist, variants=variants)
+
+        item = {
+            "distribution": dist,
+            "package": package,
+        }
+
+        if exists(package, packagesdir):
+            existing.append(item)
+        else:
+            new.append(item)
+
+    if not new:
+        return []
+
+    for item in new:
+        deploy(item["distribution"], item["package"], path=packagesdir)
+
+    shutil.rmtree(tempdir)
+    return [item["package"] for item in new]
+
+
+def download(names, tempdir=None, no_deps=False):
+    """Gather pip packages in `tempdir`
+
+    Arguments:
+        names (list): Names of packages to install, in pip-format,
+            e.g. ["six==1"]
+        tempdir (str, optional): Absolute path to where pip packages go until
+            they've been installed as Rez packages, defaults to the cwd
+        no_deps (bool, optional): Equivalent to pip --no-deps, default to False
+
+    Returns:
+        distributions (list): Downloaded distlib.database.InstalledDistribution
+
+    Raises:
+        OSError: On anything gone wrong with subprocess and pip
+
+    """
+
+    assert isinstance(names, (list, tuple)), (
+        "%s was not a tuple or list" % names
+    )
+    assert all(isinstance(name, _basestring) for name in names), (
+        "%s contained non-string" % names
+    )
+
+    tempdir = tempdir or os.getcwd()
+
+    # Build pip commandline
+    cmd = [
+        "python", "-m", "pip", "install",
+        "--target", tempdir,
+
+        # Only ever consider wheels, anything else is ancient
+        "--use-pep517",
+
+        # Handle case where the Python distribution used alongside
+        # pip already has a package installed in its `site-packages/` dir.
+        "--ignore-installed",
+
+        # rez pip users don't have to see this
+        "--disable-pip-version-check",
+
+        # Prevent user-settings from interfering with install
+        "--isolated",
```

Comment:
Some studios don't have internet and uses a global config on machines or environments to points to a local PyPI index. This option will break this. What was the reasoing being using `--isolated`?

Reply:
Oh. The reason was to reduce the room for error, and the number of tests we'd need to write. I figure it wouldn't hurt, but I did not consider this usecase. Will remove it.